primitives/tf_encrypted/primitives/paillier/primitives.py

```diff
@@ -35,49 +50,52 @@ def __init__(self, p, q):
         self.d2 = tf_big.inv(order_of_n, self.n)
         self.e = tf_big.inv(self.n, order_of_n)
 
-    def export(self, dtype: tf.DType = tf.string):
+    def export(self, dtype: tf.DType = tf.uint8) -> Tuple[tf.Tensor, tf.Tensor]:
         return (
-            tf_big.convert_from_tensor(self.p, dtype=dtype),
-            tf_big.convert_from_tensor(self.q, dtype=dtype),
+            _export_maybe_limbs(self.p, dtype),
+            _export_maybe_limbs(self.q, dtype),
         )
 
 
-def gen_keypair(bitlength=2048):
+def gen_keypair(bitlength=2048) -> Tuple[EncryptionKey, DecryptionKey]:
     p, q, n = tf_big.random_rsa_modulus(bitlength=bitlength)
     ek = EncryptionKey(n)
     dk = DecryptionKey(p, q)
     return ek, dk
 
 
 class Randomness:
-    def __init__(self, raw_randomness):
-        self.raw = tf_big.convert_to_tensor(raw_randomness)
+    def __init__(self, raw_randomness: tf.Tensor):
+        self.raw = _import_maybe_limbs(raw_randomness)
 
-    def export(self, dtype: tf.DType = tf.string):
-        return tf_big.convert_from_tensor(self.raw, dtype=dtype)
+    def export(self, dtype: tf.DType = tf.string) -> tf.Tensor:
+        return _export_maybe_limbs(self.raw, dtype=dtype)
 
 
-def gen_randomness(ek, shape):
+def gen_randomness(ek: EncryptionKey, shape) -> Randomness:
     return Randomness(tf_big.random_uniform(shape=shape, maxval=ek.n))
 
 
 class Ciphertext:
-    def __init__(self, ek: EncryptionKey, raw_ciphertext):
+    def __init__(self, ek: EncryptionKey, raw_ciphertext: tf.Tensor):
         self.ek = ek
-        self.raw = tf_big.convert_to_tensor(raw_ciphertext)
+        self.raw = _import_maybe_limbs(raw_ciphertext)
 
-    def export(self, dtype: tf.DType = tf.string):
-        return tf_big.convert_from_tensor(self.raw, dtype=dtype)
+    def export(self, dtype: tf.DType = tf.string) -> tf.Tensor:
```

Comment:
any reason why `tf.string` is used here and `tf.uint8` is used above?

Reply:
nice catch! no, we should probably use tf.unit8 everywhere, will push a fix